File: client/src/types/media.ts
import type { MediaTagResponse } from './media-tag'

export type MediaType = 'image' | 'audio' | 'book' | 'video'

export interface MediaTypeInfo {
    id: string,
    name: string,
    slug: string,
    icon: string,
    color: string
}

export interface MediaCreate {
    slug: string,
    title: string,
    filePath: string,
    fileSize: number,
    mimeType: string,
    mediaType: MediaType,
    metaData?: Record<string, any>,
    tagIds?: string[]
}

export interface MediaUpdate {
    title?: string,
    metaData?: Record<string, any>,
    tagIds?: string[]
}

export interface MediaResponse {
    id: string
    slug: string
    title: string
    filePath: string    
    fileSize: number
    mediaType: MediaType
    mimeType: string
    extension: string
    metaData: Record<string, any>
    uploadedAt: string
    tags: MediaTagResponse[]
}

export interface MediaUploadResponse {
    filePath: string
    fileSize: number
    mimeType: string
    extension: string
    mediaType: MediaType
}

File: archive/server-v0/app/schemas/media.py
from pydantic import BaseModel, Field
from app.schemas.media_tag import MediaTagResponse
from typing import Any
from datetime import datetime


class MediaCreate(BaseModel):
    """
    创建媒体
    - file_path 由上传接口返回
    - metadata_ 按媒体类型动态填写
    """
    slug: str
    title: str
    file_path: str = Field(..., alias="filePath")
    file_size: int = Field(..., alias="fileSize")
    mime_type: str = Field(..., alias="mimeType")
    media_type: str = Field(..., alias="mediaType")
    meta_data: dict[str, Any] = Field(default_factory=dict, alias="metaData")
    tag_ids: list[str] = Field(default_factory=list, alias="tagIds")

    model_config = {"populate_by_name": True}


class MediaUpdate(BaseModel):
    """更新媒体（全字段可选）"""
    title: str | None = None
    meta_data: dict[str, Any] | None = Field(None, alias="metaData")
    tag_ids: list[str] | None = Field(None, alias="tagIds")

    model_config = {"populate_by_name": True}


class MediaResponse(BaseModel):
    """媒体详情响应"""
    id: str
    slug: str
    title: str
    file_path: str = Field(..., alias="filePath")
    file_size: int = Field(..., alias="fileSize")
    media_type: str = Field(..., alias="mediaType")
    mime_type: str = Field(..., alias="mimeType")
    extension: str
    meta_data: dict[str, Any] = Field(default_factory=dict, alias="metaData")
    uploaded_at: datetime = Field(..., alias="uploadedAt")
    tags: list["MediaTagResponse"]

    model_config = {
        "from_attributes": True,
        "populate_by_name": True
    }


class MediaUploadResponse(BaseModel):
    file_path: str = Field(..., alias="filePath")
    file_size: int = Field(..., alias="fileSize")
    mime_type: str = Field(..., alias="mimeType")
    extension: str
    media_type: str = Field(..., alias="mediaType")
    metadata: dict = {}

    model_config = {"populate_by_name": True}
